timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/IsolatedPaxosTimeLockServerIntegrationTest.java

```diff
@@ -0,0 +1,145 @@
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.File;
+
+import javax.net.ssl.SSLSocketFactory;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.common.base.Optional;
+import com.palantir.atlasdb.http.AtlasDbHttpClients;
+import com.palantir.atlasdb.timelock.paxos.PaxosTimeLockConstants;
+import com.palantir.leader.PingableLeader;
+import com.palantir.lock.RemoteLockService;
+import com.palantir.paxos.PaxosAcceptor;
+import com.palantir.paxos.PaxosLearner;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampService;
+
+import feign.RetryableException;
+import io.dropwizard.testing.ResourceHelpers;
+
+/**
+ * This test creates a single TimeLock server that is configured in a three node configuration.
+ * Since it has no quorum, timestamp and lock requests (and fast-forward) should fail.
+ * However it should still be pingable and should be able to participate in Paxos as well.
+ */
+public class IsolatedPaxosTimeLockServerIntegrationTest {
+    private static final String CLIENT = "test";
+
+    private static final Optional<SSLSocketFactory> NO_SSL = Optional.absent();
+
+    private static final File TIMELOCK_CONFIG_TEMPLATE =
+            new File(ResourceHelpers.resourceFilePath("paxosThreeServers.yml"));
+
+    private static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+    private static final TemporaryConfigurationHolder TEMPORARY_CONFIG_HOLDER =
+            new TemporaryConfigurationHolder(TEMPORARY_FOLDER, TIMELOCK_CONFIG_TEMPLATE);
+    private static final TimeLockServerHolder TIMELOCK_SERVER_HOLDER =
+            new TimeLockServerHolder(TEMPORARY_CONFIG_HOLDER::getTemporaryConfigFileLocation);
+
+    @ClassRule
+    public static final RuleChain ruleChain = RuleChain.outerRule(TEMPORARY_FOLDER)
+            .around(TEMPORARY_CONFIG_HOLDER)
+            .around(TIMELOCK_SERVER_HOLDER);
+
+    @Test
+    public void cannotIssueTimestampsWithoutQuorum() {
+        assertThatThrownBy(() -> getTimestampService(CLIENT).getFreshTimestamp())
+                .satisfies(IsolatedPaxosTimeLockServerIntegrationTest::isRetryableExceptionWhereLeaderCannotBeFound);
+    }
+
+    @Test
+    public void cannotIssueLocksWithoutQuorum() {
+        assertThatThrownBy(() -> getLockService(CLIENT).currentTimeMillis())
+                .satisfies(IsolatedPaxosTimeLockServerIntegrationTest::isRetryableExceptionWhereLeaderCannotBeFound);
+    }
+
+    @Test
+    public void cannotPerformTimestampManagementWithoutQuorum() {
+        assertThatThrownBy(() -> getTimestampManagementService(CLIENT).fastForwardTimestamp(1000L))
+                .satisfies(IsolatedPaxosTimeLockServerIntegrationTest::isRetryableExceptionWhereLeaderCannotBeFound);
```

Comment:
Yep. I made this static as it doesn't really concern a given instance of the class, though agree that it reads better this way - changed.